service/src/main/java/com/shortlink/service/server/UrlConversionServer.java

```diff
@@ -1,19 +1,101 @@
 package com.shortlink.service.server;
 
+import com.shortlink.service.data.UrlMap;
+import com.shortlink.service.data.UrlMapRepository;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.cache.annotation.Cacheable;
+import org.springframework.data.redis.core.RedisTemplate;
 import org.springframework.stereotype.Component;
 
 @Component
 public class UrlConversionServer {
+    @Autowired
+    private RedisTemplate<String, Object> redisTemplate;
+
+    @Autowired
+    private UrlMapRepository urlRepository;
+
     RandomLtS randomLtS = new RandomLtS();
     Base62LtS base62LtS = new Base62LtS();
 
+    public void addToCache(String key, String value) {
+        redisTemplate.opsForValue().set(key, value);
+    }
+    public Object findInCache(String key) {
+        return redisTemplate.opsForValue().get(key);
+    }
+
+    @Cacheable("urlMapping")
     public String randomLtS(String longUrl){
```

Comment:
The cache and DB logic should be inside the Service

Reply:
The cache and DB logic are moved into "Services"